float64 left_s0
float64 left_s1
float64 left_e0
float64 left_e1
float64 right_s0
float64 right_s1
float64 right_e0
float64 right_e1